src/transformStateWithClones.js

```diff
@@ -8,6 +8,37 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let currentState = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    let newState = { ...currentState };
+
+    switch (action.type) {
+      case 'clear':
+        newState = {};
+        break;
+      case 'addProperties':
+        if (typeof action.extraData === 'object' && action.extraData !== null) {
+          Object.entries(action.extraData).forEach(([key, value]) => {
+            newState[key] = value;
+          });
+        }
```

Comment:
[CODE STYLE] - Since you are already creating a new `newState` object from `currentState`, you can simplify the addition of properties by using the object spread operator instead of `Object.entries` and a `forEach` loop. This will make your code cleaner and more efficient.